test/core/test_poll.c

```diff
@@ -26,7 +47,12 @@ int main() {
   printf("ret: %d\n", poll(multi, 5, 123));
   printf("errno: %d\n", errno);
   printf("multi[0].revents: %d\n", multi[0].revents == (POLLIN | POLLOUT));
+#if WASMFS
+  // The size of the device is 0 in WasmFS, so the device cannot POLLIN.
+  printf("multi[1].revents: %d\n", multi[1].revents == POLLOUT);
```

Comment:
I realized that I forgot to add a comment here, but it looks like `__syscall_poll` checks if a device can POLLIN based on the size of the file. 

Since `getSize()` from JSIMPL wasn't in the original API, my current workaround is to add a default `getSize: () => {}` method when instantiating a device if the user doesn't pass `getSize` to avoid an error for empty devices. Would it make sense for the default method to return a non zero value so `POLLIN` is true (For full compatibility)?

Reply:
Sorry I missed this comment before. Good point about the size...

I'm worried about returning a non-zero value, as that could cause other issues, though my worry is pretty vague.

I lean towards landing this PR as is, maybe just with a TODO on line 51 to mention the issue. @tlively what do you think?